compiler/rustc_trait_selection/src/regions.rs

```diff
@@ -24,12 +24,13 @@ impl<'tcx> InferCtxtRegionExt<'tcx> for InferCtxt<'tcx> {
             let ty = self.resolve_vars_if_possible(ty);
 
             if self.next_trait_solver() {
-                crate::solve::deeply_normalize(
+                crate::solve::deeply_normalize_with_skipped_universes(
```

Comment:
probably nicer to change `process_registered_region_obligations` to first rebind and then normalize instead of accepting skipped universes here?

Reply:
This is somewhat annoying, since then the `deeply_normalize` predicate needs to take a `PolyTypeOutlivesPredicate` here. When processing region obligations, these type outlives obligations are *not* bound, so we'd need to bind them up, and then skip their binder again. It just feels redundant :/

I'd rather do something like pass the outermost exclusive binder as part of normalize so we ICE if we find bound vars further than `ty::INNERMOST`, and only when normalizing the param-envs.

Reply:
> When processing region obligations, these type outlives obligations are not bound, so we'd need to bind them up, and then skip their binder again. It just feels redundant :/

I would prefer doing so and using `ty::Binder::dummy(sup_type)` when normalizing region obligations :thinking: we can talk about this in sync, but I strongly dislike having escaping bound vars anywhere. And believe that having to think about escaping bound vars adds unnecessary mental complexity. While this is mostly vibe based, I do feel fairly strongly about this.